Signal/src/ViewControllers/GifPicker/GifPickerCell.swift

```diff
@@ -0,0 +1,161 @@
+//
+//  Copyright (c) 2017 Open Whisper Systems. All rights reserved.
+//
+
+import Foundation
+
+class GifPickerCell: UICollectionViewCell {
+    let TAG = "[GifPickerCell]"
+
+    // MARK: Properties
+
+    var imageInfo: GiphyImageInfo? {
+        didSet {
+            AssertIsOnMainThread()
+
+            ensureLoad()
+        }
+    }
+
+    var shouldLoad = false {
```

Comment:
Loading and playing GIFs is quite expensive (network, memory, cpu).  Here's a bit of logic to _not_ preload offscreen cells that are prefetched.  I might go a step farther and stop playback of cells that scroll offscreen.